plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1831,14 +1863,23 @@ private void prepareDropPartition(SchemaTableName schemaTableName, List<String>
         {
             metastoreDeleteOperations.add(new IrreversibleMetastoreOperation(
                     format("drop partition %s.%s %s", schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionValues),
-                    () -> delegate.dropPartition(schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionValues, deleteData)));
+                    () -> {
+                        Optional<Partition> droppedPartition = delegate.getPartition(schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionValues);
+                        try {
+                            delegate.dropPartition(schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionValues, deleteData);
+                        }
+                        finally {
+                            droppedPartition.ifPresent(tableInvalidationCallback::invalidate);
```

Comment:
in other prepare... methods you call `partitionsToInvalidate.add`.
is this different intentionally?

Reply:
https://github.com/trinodb/trino/pull/10915/files/17178c24179933b6f2e9ca3340874e9272d176e1#r810075842